Add job from new layout with ci-bootstrap on main branch

viroel opened the pull request:
This job is the same as the staging branch one, but has a different parent defined in config repo[1],
which sets the ci-bootstrap main branch to be used.

The job based on ci-bootstrap repo should replace the code that we use from config repos, and allow us to test it as a role in its own repo. This job also takes a new network definition as input to configure the infrastructure needed to run our tests on zuul.
This patch doesn't wire up, or replace the current multinode-edpm job, but may do this in a next PR.

[1] https://review.rdoproject.org/r/c/config/+/51257/2/zuul.d/_jobs-crc.yaml

As a pull request owner and reviewers, I checked that:
- [X] Appropriate testing is done and actually running
- [X] Appropriate documentation exists and/or is up-to-date:
  - [X] README in the role
  - [X] Content of the docs/source is reflecting the changes

marios reviewed (commented):
lgtm but we need the depends-on to merge first

marios commented on zuul.d/edpm_multinode.yaml:
```diff
@@ -292,6 +292,16 @@
 # Jobs using ci-bootstrap layout
 #
+- job:
+    name: podified-multinode-edpm-deployment-crc
```
where is it wired up - will you add it to the github-check run in this patch?

marios reviewed (commented):
lgtm and depends-on merged

but will you wire up the job here? or are we going to wait for testproject test on this before merge?

viroel commented:
recheck

viroel commented:
> lgtm and depends-on merged
> 
> but will you wire up the job here? or are we going to wait for testproject test on this before merge?

not really, this job is the same as the one wire up here https://github.com/openstack-k8s-operators/ci-framework/pull/1020
now that we moved all ci-bootstrap content from staging branch to main branch.
The difference between these 2 is only the branch that we do the checkout for ci-bootstrap

cjeanner commented:
recheck

cjeanner commented:
/approve

lewisdenny commented:
recheck

viroel commented:
recheck

marios commented:
> > lgtm and depends-on merged
> > but will you wire up the job here? or are we going to wait for testproject test on this before merge?
> 
> not really, this job is the same as the one wire up here #1020 now that we moved all ci-bootstrap content from staging branch to main branch. The difference between these 2 is only the branch that we do the checkout for ci-bootstrap

i don't follow, i mean pull 1020 wires up some new pre-run plays for cifmw-extracted-crc-pre-bootstrap

my question was where are we going to run this new job you are adding here podified-multinode-edpm-deployment-crc but I see it is running in this PR there https://review.rdoproject.org/zuul/buildset/1efb03985d8049d591de5ac2c09afc3a

However I just checked and it looks like there are runs for this job from yesterday in other repos (edpm-ansible/install_yamls/dataplane-operator)  22nd but I don't see how if we are just adding it here. Did those run with depends-on? https://review.rdoproject.org/zuul/builds?job_name=podified-multinode-edpm-deployment-crc&skip=0

viroel commented:
> > > lgtm and depends-on merged
> > > but will you wire up the job here? or are we going to wait for testproject test on this before merge?
> > 
> > 
> > not really, this job is the same as the one wire up here #1020 now that we moved all ci-bootstrap content from staging branch to main branch. The difference between these 2 is only the branch that we do the checkout for ci-bootstrap
> 
> i don't follow, i mean pull 1020 wires up some new pre-run plays for cifmw-extracted-crc-pre-bootstrap
> 
> my question was where are we going to run this new job you are adding here podified-multinode-edpm-deployment-crc but I see it is running in this PR there https://review.rdoproject.org/zuul/buildset/1efb03985d8049d591de5ac2c09afc3a
> 
> However I just checked and it looks like there are runs for this job from yesterday in other repos (edpm-ansible/install_yamls/dataplane-operator) 22nd but I don't see how if we are just adding it here. Did those run with depends-on? https://review.rdoproject.org/zuul/builds?job_name=podified-multinode-edpm-deployment-crc&skip=0

First of all, sorry Marios for not paying attention in your question. You are correct when asking about the name of the job and all the confusion. I choose by mistake the same name of the current multinode-edpm job, and that's why it was flat failing here. Thanks for pointing to that.
I changed the name to identify it as the job which uses ci-bootstrap role to configure the infra.
The idea is to replace the current edpm-multinode job, but in future PRs.

I will add testproject results as soon as reports success.

viroel commented:
/hold

testing in testproject

viroel commented:
podified-multinode-edpm-deployment-crc-bootstrap-testproject https://review.rdoproject.org/zuul/build/ceb1ae66061948bda033fc5b320d5b65 : SUCCESS in 1h 20m 47s

viroel commented:
/test pre-commit

viroel commented:
> Ok thanks for update @viroel but question still stands - where are we going to wire it up to run (on pull requests?, periodic?)

This is going to be wired up in PRs, is focused on job with CRC based jobs running in a OCP env. The difference with current job is that is using ci-bootstrap role and is based on Pablo's new network definition. But the job for now is a mix of the new layout + old deployment. So it is needs more updates.

viroel commented:
/hold cancel
